python/nav/web/auth.py

```diff
@@ -78,3 +94,39 @@ def authenticate(username, password):
         return account
     else:
         return None
+
+
+def authenticate_remote_user(request=None):
+    if not NAV_CONFIG.get('AUTH_SUPPORT_REMOTE_USER', False):
+        return None
+
+    if not request:
+        return None
+
+    username = request.META.get('REMOTE_USER', None)
+    if not username:
+        return None
+
+    # We now have a username-ish
+
+    try:
+        account = Account.objects.get(login=username)
+    except Account.DoesNotExist:
+        # Store the ldapuser in the database and return the new account
+        account = Account(
+            login=username,
+            name=username,
+            ext_sync='REMOTE_USER'
+        )
+        account.set_password(fake_password(32))
+        account.save()
+        logger.info("Created user %s from header REMOTE_USER", account.login)
```

Comment:
This also needs audit logging.

Reply:
Audit-logging added in "Audit some remote user events"